LibTrixi.jl/src/api_jl.jl

```diff
@@ -49,3 +49,60 @@ function trixi_step_jl(simstate)
 
     return nothing
 end
+
+function trixi_ndims_jl(simstate)
+    mesh, _, _, _ = mesh_equations_solver_cache(simstate.semi)
+    return ndims(mesh)
+end
+
+function trixi_nelements_jl(simstate)
+    _, _, solver, cache = mesh_equations_solver_cache(simstate.semi)
+    return nelements(solver, cache)
+end
+
+function trixi_nvariables_jl(simstate)
+    _, equations, _, _ = mesh_equations_solver_cache(simstate.semi)
+    return nvariables(equations)
+end
+
+function trixi_load_cell_averages_jl(data, simstate)
+    mesh, equations, solver, cache = mesh_equations_solver_cache(simstate.semi)
+    n_elements = nelements(solver, cache)
+    n_variables = nvariables(equations)
+    n_nodes = nnodes(solver)
+    n_dims = ndims(mesh)
+
+    u_ode = simstate.integrator.u
+    u = wrap_array(u_ode, mesh, equations, solver, cache)
+
+    # all permutations of nodes indices for arbitrary dimension
+    node_cis = CartesianIndices(ntuple(i -> n_nodes, n_dims))
+
+    # temporary storage for mean value on current element for all variables
+    u_mean = get_node_vars(u, equations, solver, node_cis[1], 1)
+
+    for element in eachelement(solver, cache)
+
+        # compute mean value using nodal dg values and quadrature
+        u_mean = zero(u_mean)
```

Comment:
Here, `u_mean` should be reset to zero.
Does this allocate a new `SVector` every time?
Even if this is true, something like `u_mean .= zero(eltype(..))` does not work. Because `SVector` is immutable?

Reply:
`SVector`s live on the stack and thus are not allocated. Thus this should be OK.